File: fix_paths.js

const fs = require('fs');
const path = require('path');

const directoryPath = String.raw`c:\Users\stach\Desktop\D2S Business Folder\WEBSITE WORKS\Insurance\INSURANCE\html`;

function fixPaths(dir) {
    fs.readdir(dir, (err, files) => {
        if (err) {
            return console.log('Unable to scan directory: ' + err);
        }

        files.forEach(function (file) {
            const filepath = path.join(dir, file);

            if (file.endsWith('.kit') || file.endsWith('.html')) {
                fs.readFile(filepath, 'utf8', (err, content) => {
                    if (err) {
                        return console.log(err);
                    }

                    let newContent = content;

                    // Fix CSS/JS paths
                    newContent = newContent.replace(/href="\/dist/g, 'href="./dist');
                    newContent = newContent.replace(/src="\/dist/g, 'src="./dist');
                    newContent = newContent.replace(/href="\/favicon.ico"/g, 'href="favicon.ico"');

                    // Fix page links
                    const pages = [
                        "index.html", "about-us.html", "news.html", "careers.html",
                        "contactus.html", "propertyInsurance.html", "businessInsurance.html",
                        "vehicleInsurance.html", "travelInsurance.html", "otherInsurance.html",
                        "liabilityInsurance.html", "privacy-policy.html", "complaints.html",
                        "news-details.html", "job-details.html"
                    ];

                    pages.forEach(page => {
                        const regex = new RegExp(`href="/${page}"`, 'g');
                        newContent = newContent.replace(regex, `href="${page}"`);
                    });

                    if (content !== newContent) {
                        console.log(`Fixing paths in: ${file}`);
                        fs.writeFile(filepath, newContent, 'utf8', (err) => {
                            if (err) console.log(err);
                        });
                    }
                });
            }
        });
    });
}

fixPaths(directoryPath);

File: fix_paths.py

import os

def fix_paths(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(".kit") or file.endswith(".html"):
                filepath = os.path.join(root, file)
                try:
                    with open(filepath, 'r', encoding='utf-8') as f:
                        content = f.read()
                    
                    # Fix CSS/JS paths
                    new_content = content.replace('href="/dist', 'href="./dist')
                    new_content = new_content.replace('src="/dist', 'src="./dist')
                    new_content = new_content.replace('href="/favicon.ico"', 'href="favicon.ico"')
                    
                    # Fix internal links (e.g. href="/about-us.html" -> href="about-us.html")
                    # Be careful not to replace https://...
                    # We can target href="/[a-zA-Z]
                    
                    # Actually, simple string replacement for known pages is safer
                    pages = [
                        "index.html", "about-us.html", "news.html", "careers.html", 
                        "contactus.html", "propertyInsurance.html", "businessInsurance.html",
                        "vehicleInsurance.html", "travelInsurance.html", "otherInsurance.html",
                        "liabilityInsurance.html", "privacy-policy.html", "complaints.html",
                        "news-details.html", "job-details.html"
                    ]
                    
                    for page in pages:
                        new_content = new_content.replace(f'href="/{page}"', f'href="{page}"')

                    if content != new_content:
                        print(f"Fixing paths in: {file}")
                        with open(filepath, 'w', encoding='utf-8') as f:
                            f.write(new_content)
                except Exception as e:
                    print(f"Error processing {file}: {e}")

if __name__ == "__main__":
    fix_paths(r"c:\Users\stach\Desktop\D2S Business Folder\WEBSITE WORKS\Insurance\INSURANCE\html")
